### PR TITLE
Add integration tests for ODM features

### PR DESCRIPTION
**Pull request by mcfedr**
Same as #39 but without extra composer deps

Had to leave 'config.mongodb' in composer json so that it would allow installing doctrine-odm

### CODE REVIEW

**Comment by mcfedr** [comment]
The travis job has failed because of some travis issue, but you can see in the php 7.1 job that it passes.

**Comment by ondrejmirtes** [comment]
Please rebase on top of master, it conflicts now.

File: composer.json [hunk 1]
```diff
@@ -39,5 +41,10 @@
 	},
 	"autoload-dev": {
 		"classmap": ["tests/"]
+	},
+	"config": {
+		"platform": {
+			"ext-mongo": "1.6.16"
```
Comment (ondrejmirtes) [inline composer.json]: What's this for?

Comment (mcfedr) [reply composer.json]: `doctrine/mongodb-odm` requires `ext-mongo`, this tells composer that its installed, even though it isnt, and allows odm to be installed

Comment (mcfedr) [reply composer.json]: The other option is to run `composer install` with `--ignore-platform-reqs`, if you think thats better @ondrejmirtes ?

Comment (ondrejmirtes) [reply composer.json]: We can leave this in :)

**Comment by ondrejmirtes** [comment]
Thanks!

**Pull request merged**